lib/agent.js

```diff
@@ -72,6 +83,25 @@ Agent.prototype.createSocket = function (req, options) {
   return socket;
 };
 
+if (node10) {
+    Agent.prototype.removeSocket = function (s, options) {
+      OriginalAgent.prototype.removeSocket.call(this, s, options);
+      var name = this.getName(options);
+      debug('removeSocket', name, 'destroyed:', s.destroyed);
+
+      if (s.destroyed && this.freeSockets[name]) {
```

Comment:
free sockets has been delete on https://github.com/lattmann/agentkeepalive/blob/master/lib/_http_agent.js#L276

use the _http_agent default `removeSocket` will be ok.